SpatialGDK/Source/SpatialGDK/Public/SpatialView/ConnectionHandlers/SpatialOSConnectionHandler.h

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "SpatialView/ConnectionHandlers/AbstractConnectionHandler.h"
+#include "SpatialView/OpList/OpList.h"
+#include "SpatialView/OpList/WorkerConnectionOpList.h"
+
+namespace SpatialGDK
+{
+
+class SpatialOSConnectionHandler : public AbstractConnectionHandler
+{
+public:
+	explicit SpatialOSConnectionHandler(Worker_Connection* Connection)
+	: Connection(Connection), WorkerId(UTF8_TO_TCHAR(Worker_Connection_GetWorkerId(Connection)))
+	{
+		const Worker_WorkerAttributes* Attributes = Worker_Connection_GetWorkerAttributes(Connection);
+		for (uint32 i = 0; i < Attributes->attribute_count; ++i)
+		{
+			WorkerAttributes.Push(FString(UTF8_TO_TCHAR(Attributes->attributes[i])));
+		}
+	}
+
+	void Advance() override
+	{
+	}
+
+	uint32 GetOpListCount() override
+	{
+		return 1;
+	}
+
+	OpList GetNextOpList() override
+	{
+		OpList Ops = GetOpListFromConnection(Connection.Get());
+		// Find responses and change their internal request IDs to match the request ID provided by the user.
+		for (uint32 i = 0; i < Ops.Count; ++i)
```

Comment:
Nit: Wonder if a for each style loop would be slightly simpler. Doesn't seem like the index is used for anything else but accessing an element.

Reply:
We'd need to give oplist and begin and end function to do that.
I'm not against doing that but it isn't there now.

Reply:
Ahh right that makes sense.